### ISSUE TITLE
Address comments from DBA review

### ISSUE THREAD

**Issue by chilts** [mozilla/fxa-auth-db-server#111]
* https://bugzilla.mozilla.org/show_bug.cgi?id=1096631

Just extracting a few points here for things we may wish to address. If we decide not to address any of these here, we'll make a new issue for later. Just listing the points for now:

* [ ] (1) procedures with just an INSERT statement (to DECIDE)
* [ ] (2) procedures that only use SELECT are fine, but hopefully you have code for if there are 0 results returned (to REVIEW)
* [ ] (3) SELECTs, you know they just run as-is, and you're not actually returning the data (to DECIDE)
* [ ] (4) comma joins (to UPDATE)
* [ ] (5) for just updates, do you want to also get the count of what was updated? check exactly one row? (to DECIDE)

(1), it is suggested to use REPLACE or INSERT IGNORE  or INSERT...ON DUPLICATE KEY UPDATE. alternatively, put in an error handler. However, I think we do actually want to return the error here so that the DB server can tell the auth server and therefore tell the client that their update failed.

(2), I *think* we already figure out in the DB server whether we have 0 or >0 results. I'll review all of these occurances.

(3) From what I can tell, the data from the select is automatically returned. I'm always happier when it's more explicit though so I'm keen on this. @rfk any strong feelings one way or the other?

(4) Yes please.

(5) Am not sure about this until I do a quick review. I thought we already had some checking though I think there are some places where we do ignore or not read how many rows were affected (my brain is wondering if this is for the DELETEs rather than the UPDATEs currently).

@rfk Your thoughts welcome or anything further I have missed from Sheeri's review?

**Comment by chilts** [comment]
And @jrgm too.

**Comment by rfk** [comment]
I think you hit all the points here; some brief comments:

(1) I agree, we're mostly inserting randomly-generated unique keys so if there's a conflict, we want to fail and fail loudly.

(2) Yes, IIRC we have some helpers that deal with this, but definitely worth reviewing.

(3) I don't really understand what's being suggested here, and what we have now seems to work.  I'd be happy to look at a more explicit version but I worry about adding unnecessary complexity.

(4) *shrug*

(5) Yes, worth a review.

**Comment by rfk** [comment]
@chilts are we happy enough with our existing stuff to just close this one out?

**Comment by chilts** [comment]
There were a few things I'd have liked to have done, but in all honesty, no-one seemed too worried. Yeah, I think we're happy enough.

In terms of (5), I think if we returned a number of affected rows we could do more in the case of single updates and/or deletes, but again we seem to err on the side of "whether we deleted 0 or 1 rows, we don't really mind so we'll return success anyway" so even doing this might not be worth it.
